From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Strokkur24 <strokkur.24@gmail.com>
Date: Tue, 17 Feb 2026 18:09:20 +0100
Subject: [PATCH] Rename brand


diff --git a/src/main/java/io/papermc/paper/ServerBuildInfoImpl.java b/src/main/java/io/papermc/paper/ServerBuildInfoImpl.java
index 74ffdc823e66fc5ec027c4b7c462382bcbfe2be2..e310733b22226a9a0bd303999a7bf288a1d40c18 100644
--- a/src/main/java/io/papermc/paper/ServerBuildInfoImpl.java
+++ b/src/main/java/io/papermc/paper/ServerBuildInfoImpl.java
@@ -30,7 +30,7 @@ public record ServerBuildInfoImpl(
     private static final String ATTRIBUTE_GIT_BRANCH = "Git-Branch";
     private static final String ATTRIBUTE_GIT_COMMIT = "Git-Commit";
 
-    private static final String BRAND_PAPER_NAME = "Paper";
+    private static final String BRAND_PAPER_NAME = "HourPaper";
 
     private static final String BUILD_DEV = "DEV";
 
